lib/channels/src/index.ts

```diff
@@ -0,0 +1,152 @@
+export type ChannelHandler<TEventArgs = any> = (event: ChannelEvent<TEventArgs>) => void;
+
+export interface ChannelTransport<TEventArgs = any> {
+  send(event: ChannelEvent<TEventArgs>): void;
+  setHandler(handler: ChannelHandler<TEventArgs>): void;
+}
+
+export interface ChannelEvent<TEventArgs = any> {
+  type: string; // eventName
+  from: string;
```

Comment:
Same goes for the `from` field similar to the `type` field, see comment above.

Reply:
Yeah should be either `manager` or `preview`